src/pyodide/internal/loadPackage.ts

```diff
@@ -12,16 +11,29 @@
 
 import { default as LOCKFILE } from "pyodide-internal:generated/pyodide-lock.json";
 import { WORKERD_INDEX_URL } from "pyodide-internal:metadata";
-import { SITE_PACKAGES, LOAD_WHEELS_FROM_R2, getSitePackagesPath } from "pyodide-internal:setupPackages";
+import {
+  SITE_PACKAGES,
+  LOAD_WHEELS_FROM_R2,
+  getSitePackagesPath,
+} from "pyodide-internal:setupPackages";
 import { parseTarInfo } from "pyodide-internal:tar";
 import { default as DiskCache } from "pyodide-internal:disk_cache";
 import { createTarFS } from "pyodide-internal:tarfs";
 
-async function decompressArrayBuffer(arrBuf) {
-  return await new Response(new Response(arrBuf).body.pipeThrough(new DecompressionStream("gzip"))).arrayBuffer();
+async function decompressArrayBuffer(
+  arrBuf: ArrayBuffer,
+): Promise<ArrayBuffer> {
+  const resp = new Response(arrBuf);
+  if (resp && resp.body) {
+    return await new Response(
+      resp.body.pipeThrough(new DecompressionStream("gzip")),
+    ).arrayBuffer();
+  } else {
+    throw new Error("Failed to decompress array buffer");
+  }
```

Comment:
I think the following code would be slightly cleaner here (though it's pretty unimportant):
```js
const blob = new Blob([arrBuf]);
const decompressed = blob.stream().pipeThrough(new DecompressionStream("gzip"));
return await new Response(decompressed).arrayBuffer();
```